metpy/xarray.py

```diff
@@ -1,7 +1,20 @@
-# Copyright (c) 2018 MetPy Developers.
+# Copyright (c) 2018,2019 MetPy Developers.
 # Distributed under the terms of the BSD 3-Clause License.
 # SPDX-License-Identifier: BSD-3-Clause
-"""Provide accessors to enhance interoperability between XArray and MetPy."""
+"""Provide accessors to enhance interoperability between xarray and MetPy.
+
+MetPy relies upon the `CF Conventions <http://cfconventions.org/>`_. to provide helpful
+attributes and methods on xarray DataArrays and Dataset for working with
+coordinate-related metadata. Also included are several attributes and methods for unit
+operations.
+
+These accessors will be activated with any import of MetPy. Do not use the
+``MetPyDataArrayAccessor`` or ``MetPyDatasetAccessor`` classes directly, instead, utilize the
+applicable properties and methods via the ``.metpy`` attribute on an xarray DataArray or
+Dataset.
+
+See Also: :doc:`xarray with MetPy Tutorial </tutorials/xarray_tutorial>`.
+"""  # noqa: RST304
```

Comment:
I think the `noqa` needs to be moved up a line in order for pytest to catch it.

Reply:
Would you be able to clarify what you mean by moved up a line? Should it be before the multiline string?

Reply:
CI failed on recognizing it during the last attempt...we'll see what it does this time. I'm wondering if it should be within the string, just a line up at the end of the line after the `:doc:` link.

Reply:
I've pushed a commit to see if that makes CI happy, but unfortunately that puts the `# noqa: RST304` as text in the docs themselves, which is not ideal.

Reply:
Nope, that didn't seem to do it. Based on all the googling I did about it, having it right after the closing quotes on the same line looks like it *should* work...but it's not. Also, for whatever odd reason, I can't get the RST checks working for me locally, so I haven't been able to play around with it except pushing updates and waiting for Travis.

Would you happen to have any other ideas?

Reply:
Have you tried moving the noqa on the same line as the :doc: reference? That's what I originally meant about moving it up.

Reply:
I finally got the RST checks working locally, and after trying that and the other options again, unfortunately any comments inside the multiline string are just included as part of the string and the noqa has no effect. The only workaround I found was adding RST304 to the per-file-ignores in setup.cfg.

Reply:
Also, might https://gitlab.com/pycqa/flake8/issues/375 be related?

Reply:
Well, that certainly looks related. Thanks for digging into it and resolving this. Hopefully all passes now.